From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Paul Gofman <pgofman@codeweavers.com>
Date: Fri, 6 Mar 2026 17:28:32 -0600
Subject: [PATCH] secur32: HACK: Add %NO_SHUFFLE_EXTENSIONS to priority string.

To workaround GnuTLS bug (preventing handshake with certain servers)
present in 3.8.9 and fixed by dc5ee80c3a28577e9de0f82fb08164e4c02b96af).

CW-Bug-Id: #26715
Valve-commit: cb50ee2f25521aa7a11b07c3fcedcf9db73798a5
Picked: clean
Class: valve
Evidence: docs/evidence/2026-09-28-wine-proton-rebase.md
Offered-upstream: no
---
 dlls/secur32/schannel_gnutls.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/secur32/schannel_gnutls.c b/dlls/secur32/schannel_gnutls.c
index 7d0db52..bb7d8fb 100644
--- a/dlls/secur32/schannel_gnutls.c
+++ b/dlls/secur32/schannel_gnutls.c
@@ -434,7 +434,7 @@ static int pull_timeout(gnutls_transport_ptr_t transport, unsigned int timeout)
 
 static NTSTATUS set_priority(schan_credentials *cred, gnutls_session_t session)
 {
-    char priority[128] = "NORMAL:%LATEST_RECORD_VERSION", *p;
+    char priority[160] = "NORMAL:%LATEST_RECORD_VERSION:%NO_SHUFFLE_EXTENSIONS", *p;
     BOOL server = !!(cred->credential_use & SECPKG_CRED_INBOUND);
     const struct protocol_priority_flag *protocols =
         server ? server_protocol_priority_flags : client_protocol_priority_flags;
